src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -89,7 +89,11 @@ public virtual Stream GetNuspec()
         {
             // This is the default implementation. It is overridden and optimized in
             // PackageArchiveReader and PackageFolderReader.
+            return GetStream(GetNuspecFile());
+        }
 
+        public virtual string GetNuspecFile()
+        {
             // Find all nuspecs in the root folder.
             var nuspecPaths = GetFiles()
                 .Where(entryPath => PackageHelper.IsManifest(entryPath))
```

Comment:
I looked into that, there was no reason for the old code to do this that I can find.

GetNuspecFile was used in two places: PackageExtractor where it passes the path into a ZipArchive for the source, and in PackageArchiveReader. In both scenarios the forward slash is the correct syntax, not what this extension method was doing.

Within PackageFolderReader where this could be a problem it uses FileInfo to handle this inside the GetFile method which abstracts the path differences between folders and zips.